src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -281,6 +294,114 @@ public synchronized Response modifyUsersRole(@Context final HttpServletRequest r
     return Response.ok().build();
   }
 
+  /**
+   * This method will allow users to have their teacher pending status mass changed.
+   *
+   * @param request - to help determine access rights.
+   * @param status  - new teacher pending status.
+   * @param userIds - a list of user ids to change en-mass
+   * @return Success shown by returning an ok response
+   */
+  @POST
+  @Path("/users/change_teacher_pending/{status}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Update teacher_pending to true or false for a list of possible user IDs.",
+      description = "This endpoint requires an admin user to be logged in. If updating teacher_pending to false it "
+          + "also sends an email to decline teacher account upgrade. If email send fails, response is still OK but "
+          + "with added message to notify user")
+  public synchronized Response modifyUsersTeacherPendingStatus(@Context final HttpServletRequest request,
+                                                               @PathParam("status") final Boolean status,
+                                                               final List<Long> userIds) {
+
+    Map<String, List<Long>> failedUpdates = new HashMap<>();
+
+    RegisteredUserDTO requestingUser;
+
+    try {
+      requestingUser = userManager.getCurrentRegisteredUser(request);
+      if (!isUserAnAdminOrEventManager(userManager, requestingUser)) {
+        return new SegueErrorResponse(Status.FORBIDDEN, ACCESS_DENIED_MESSAGE)
+            .toResponse();
+      }
+    } catch (NoUserLoggedInException e) {
+      return SegueErrorResponse.getNotLoggedInResponse();
+    }
+
+    for (Long userId : userIds) {
+      try {
+        modifyTeacherPendingStatusForUser(userId, requestingUser, status, failedUpdates);
+      } catch (SegueDatabaseException e) {
+        log.error("Database error while trying to change teacher_pending status", e);
+        return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR,
+            "Could not update teacher_pending status").toResponse();
+      }
+    }
+
+    if (failedUpdates.containsKey(USERS_NOT_FOUND)) {
+      String errorMessage =
+          String.format("One or more users could not be found: %s", failedUpdates.get(USERS_NOT_FOUND));
+      if (failedUpdates.containsKey(FAILED_TO_SEND)) {
+        errorMessage += String.format(" Emails could not be sent to userIds: %s", failedUpdates.get(FAILED_TO_SEND));
+      }
+      return new SegueErrorResponse(Status.BAD_REQUEST, errorMessage).toResponse();
+    }
+
+    String responseString = "Teacher pending status updated to " + status + " for requested userIds: " + userIds;
```

Comment:
Small one, but it looks like this could be part of an if/else, since the response is always going to be either this one or the one below. The end result is the same, but by making it part of an if/else, it's clearer. Also, `responseMessage` is a bit more meaningful.